test/test_timesteparrays.jl

```diff
@@ -282,27 +277,49 @@ data = collect(reshape(1:64, 4, 4, 4))
 arr_fixed = TimestepArray{FixedTimestep{2000, 5}, Int, 3, 1}(data)
 arr_variable = TimestepArray{VariableTimestep{years}, Int, 3, 1}(data)
```

Comment:
Also, are there test any where of the colon as a range functionality?

Reply:
That's a good point, and I should probably do that I've been kicking the can down the road a little bit because I wasn't quite sure how to move the time index. In your code above what is indicating the location of the `ti`?  I put some tests of colon functionality into `time.jl`, and they are tested in our `mcs` testing as well, but I should add some more explicit ones.

Reply:
The `ti` is just the fourth type parameter of the `TimestepArray`, so I just changed the first one in `arr_fixed` to 2 and the second one in `arr_variable` to 3. So if we make that change here, then in all the following tests of getindex and setindex, you would have to pass the TimestepIndex or TimestepValue as the second or third index, instead of the first one, to test the functionality

Reply:
For the colon stuff, I saw tests of using Colon for the whole dimension, so that's covered (both in the `mcs` stuff and explicitly in a couple test files) but what I meant was that we could also add some tests for using the colon for a range like: `arr[TimetsepIndex(3):TimestepIndex(10), 1]` since you also added methods for that and I think that one doesn't get tested by `mcs`? Or does it?

Reply:
Ah ok got it thanks!  Per the colons, I had sort of split up the tests so that in `test_timesteps.jl` I checked that
`TimestepIndex(10): TimestepIndex(15) == TimestepIndex(10):1: TimestepIndex(15) == [TimestepIndex(10)... ` etc.
and then in `test_timesteparrays` I tried indexing with an array.  But I can add some more.

Reply:
Yea I guess I sort of added that in `test_timesteps.jl` as a unit test but I think you're right we should add a full test of it to this file too.

Reply:
Ah cool that seems sufficient to me!